packages/common/src/search/types/IFilter.ts

```diff
@@ -0,0 +1,123 @@
+import { AccessLevel } from "../..";
+import { IMatchOptions, IDateRange, IRelativeDate } from "./types";
+
+/**
+ * Filter with properties for well-known filterable fields, across all searchable entities - items, users, groups, events.
+ * Fields which do not apply to a given type are simply ignored.
+ *
+ * Used with `IFilterGroup` to construct complex queries or on it's own for simpler queries.
+ */
+export interface IFilter {
+  /**
+   * Applies to Item and Group
+   */
+  access?: AccessLevel | AccessLevel[] | IMatchOptions;
+  /**
+   * Applies to Item, Group and User
+   */
+  created?: IDateRange<number> | IRelativeDate;
+  /**
+   * Applies to Item and Group
+   */
+  description?: string | string[] | IMatchOptions;
+  /**
+   * Applies to User
+   */
+  disabled?: boolean;
+  /**
+   * Applies to User
+   */
+  email?: string | string[] | IMatchOptions;
+  /**
+   * Applies to User
+   */
+  firstname?: string | string[] | IMatchOptions;
+  /**
+   * Applies to User
+   */
+  fullname?: string | string[] | IMatchOptions;
+  /**
+   * Applies to Item
+   */
+  group?: string | string[] | IMatchOptions;
+  /**
+   * Applies to User
+   */
+  groups?: string | string[] | IMatchOptions;
```

Comment:
Because the backing API's use different property names... we can look at unifying but that will require some finicky special case coding to rename the prop, but only for one of the search fns.

Reply:
Looked into this more, and `groups` is not a filterable property on anything... (according to the docs). So I'll remove that, and note that it's only applicable to Items.